### PR TITLE
Llin/ent 5438

### PR DESCRIPTION
**Pull request by long74100**
- Track life cycle events for licenses and subsidy request configs
- Update serializers so that we're returning lms_user_id & reviewer_lms_user_id instead
- Clean ups

https://openedx.atlassian.net/browse/ENT-5438

### CODE REVIEW

File: docs/segment_events.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+Segment events
```
Comment (macdiesel) [inline docs/segment_events.rst:1]: #evented

File: enterprise_access/apps/api/tasks.py [hunk 2]
```diff
@@ -169,15 +189,13 @@ def assign_licenses_task(license_request_uuids, subscription_uuid):
     license_requests = LicenseRequest.objects.filter(
         uuid__in=license_request_uuids,
         state__in=[SubsidyRequestStates.PENDING, SubsidyRequestStates.ERROR]
-    )
+    ).select_related('user')
```
Comment (macdiesel) [inline enterprise_access/apps/api/tasks.py:192]: #noN+1

**Review by macdiesel — APPROVED** [review]
This looks fine. Could you add a note to the readme file you have chosen to instrument the API end points and not the database hooks?

**Review by adamstankiewicz — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**